apps/studio/components/interfaces/Reports/ReportBlock/ReportBlockContainer.tsx

```diff
@@ -39,35 +41,28 @@ export const ReportBlockContainer = ({
         <TooltipTrigger asChild>
           <div
             className={cn(
-              'grid-item-drag-handle flex py-1 pl-3 pr-1 items-center gap-2 z-10 shrink-0 group',
+              'grid-item-drag-handle flex py-1 pl-3 pr-1 items-center gap-2 z-10 shrink-0 group h-9',
               draggable && 'cursor-move'
             )}
           >
-            <div
-              className={cn(
-                showDragHandle && 'transition-opacity opacity-100 group-hover:opacity-0'
-              )}
-            >
-              {loading ? (
-                <Loader2
-                  size={(icon as any)?.props?.size ?? 16}
-                  className="text-foreground-lighter animate-spin"
-                />
-              ) : (
-                icon
-              )}
-            </div>
-            {showDragHandle && (
+            {showDragHandle ? (
               <div className="absolute left-3 top-2.5 z-10 opacity-0 transition-opacity group-hover:opacity-100">
                 <GripHorizontal size={16} strokeWidth={1.5} />
               </div>
+            ) : icon ? (
+              icon
+            ) : (
+              <Code size={16} strokeWidth={1.5} className="text-foreground-muted" />
             )}
-            <h3
-              title={label}
-              className="!text-xs font-medium text-foreground-light flex-1 truncate"
+            <div
+              className={cn(
+                'flex items-center gap-2 flex-1 transition-opacity',
+                showDragHandle && 'group-hover:opacity-25'
+              )}
             >
-              {label}
-            </h3>
+              <h3 className="heading-meta truncate">{label}</h3>
```

Comment:
i'd suggest we don't uppercase the label here since its based on a user input (snippet title) as it could be confusing to the user

this is a bad example but for e.g if the user has 2 snippets named
- "Users table"
- "users table"


<img width="241" height="61" alt="image" src="https://github.com/user-attachments/assets/944fac3b-ab8d-4372-893f-25ac91fc9ded" />